src/store/state.js

```diff
@@ -0,0 +1,67 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const state = createSlice({
+  name: 'state',
+  initialState: {
+    activityIndex: 0,
+    adIndex: 0,
+    boardMessageIndex: 0,
+    availableQuotes: [],
+    usedQuotes: [],
+    quoteIndex: 0,
+    current: 'activities'
+  },
+  reducers: {
+    incrementActivityIndex(state) {
+      state.activityIndex++;
+    },
+    resetActivityIndex(state) {
+      state.activityIndex = 0;
+    },
+
+    incrementAdIndex(state) {
+      state.adIndex++;
+    },
+    resetAdIndex(state) {
+      state.adIndex = 0;
+    },
+
+    incrementBoardMessageIndex(state) {
+      state.boardMessageIndex++;
+    },
+    resetBoardMessageIndex(state) {
+      state.boardMessageIndex = 0;
+    },
+
+    nextQuote(state) {
+      state.usedQuotes.push(state.quoteIndex);
+
+      const hi = state.availableQuotes.length - 1;
+      const availableQuoteIndex = Math.floor(Math.random() * hi);
+
+      [state.quoteIndex] = state.availableQuotes.splice(availableQuoteIndex, 1);
+    },
+    resetQuotes(state, action) {
+      state.availableQuotes = new Array(action.payload)
+          .fill(0)
```

Comment:
What is your though process about this? What are trying to do?

Reply:
The quotes should be drawn in a random order, so the state keeps track of which indices have been used and which is currently in use and when the list is empty, it generates a new list of `action.payload` length (where the number of quotes is the payload) with indices.

Reply:
I have the feeling the logic can be easier, as you fill a whole array with 0.

Reply:
If you leave out the `.fill(0)` then the array remains empty: <img width="406" alt="Screenshot 2024-08-07 at 18 09 43" src="https://github.com/user-attachments/assets/2c5ef332-fbd5-4ce8-a61c-59bb3758bea2">